File: SSUMobile/Modules/Debug/SSUDebugModule.m
//
//  SSUDebugModule.m
//  SSUMobile
//
//  Created by Eric Amorde on 9/17/15.
//  Copyright (c) 2015 Sonoma State University Department of Computer Science. All rights reserved.
//

#import "SSUDebugModule.h"

@implementation SSUDebugModule

+ (instancetype) sharedInstance {
    static SSUDebugModule * instance = nil;
    static dispatch_once_t token;
    dispatch_once(&token, ^{
        instance = [[self alloc] init];
    });
    return instance;
}

#pragma mark - SSUModule

- (nonnull NSString *) title {
    return @"Debug";
}

- (nonnull NSString *) identifier {
    return @"debug";
}

- (UIView *) viewForHomeScreen {
    return [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"debug_icon"]];
}

- (UIImage *) imageForHomeScreen {
    return [UIImage imageNamed:@"debug_icon"];
}

- (UIViewController *) initialViewController {
    UIStoryboard * storyboard = [UIStoryboard storyboardWithName:@"Debug"
                                                          bundle:[NSBundle bundleForClass:[self class]]];
    return [storyboard instantiateInitialViewController];
}

- (BOOL) showModuleInNavigationBar {
    return NO;
}

- (BOOL) shouldNavigateToModule {
    return YES;
}

@end

File: SSUMobile/Modules/Debug/SSUDebugModule.swift
//
//  SSUDebugModule.swift
//  SSUMobile
//
//  Created by Eric Amorde on 4/13/17.
//  Copyright © 2017 Sonoma State University Department of Computer Science. All rights reserved.
//

import Foundation
import UIKit

class SSUDebugModule: SSUModuleBase, SSUModuleUI {
    
    @objc(sharedInstance)
    static let instance = SSUDebugModule()
    
    // MARK: SSUModule
    
    var title: String {
        return "Debug"
    }
    
    var identifier: String {
        return "debug"
    }
    
    func setup() {
        
    }
    
    // MARK: SSUModuleUI
    
    func imageForHomeScreen() -> UIImage? {
        return UIImage(named: "debug_icon")
    }
    
    func initialViewController() -> UIViewController {
        let storyboard = UIStoryboard(name: "Debug", bundle: Bundle(for: type(of: self)))
        return storyboard.instantiateInitialViewController()!
    }
    
    func shouldNavigateToModule() -> Bool {
        return true
    }
}
